core/kernel/notif.c

```diff
@@ -263,6 +263,25 @@ void notif_it_set_mask(unsigned int id, bool do_mask)
 		set_it_mask(id, do_mask);
 }
 
+/*
+ * Return whether interrupt notification is masked or unmasked
+ *
+ * @id Interrupt identifier
+ *
+ * Return true or false upon interrupt notif mask value
+ */
+bool notif_it_is_masked(unsigned int id)
+{
+	assert(find_notif_it(id));
+	return bit_test(notif_it_masked, id) != 0;
```

Comment:
We should test that `id` is smaller than `NOTIF_IT_VALUE_MAX` too, same below.

Reply:
this is asserted at line above.

Reply:
So this interface will only be called with an already controlled "id"?
My point is that assert catches things that "can't happen", but if they do happen it's due to a flaw inside OP-TEE only and not due to external input. If external input can cause this, then it must always be checked and we either panic() or return an error.